### PR TITLE
Support for a daemon's restart-condition

### PR DESCRIPTION
**Pull request by sergiusens**
Just makes it valid syntax in snapcraft.yaml

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by sergiusens** [comment]
retest this please

File: schema/snapcraft.yaml [hunk 1]
```diff
@@ -168,6 +168,15 @@ properties:
             enum:
               - simple
               - forking
+          restart-condition:
+              type: string
+              enum:
+                - never
```
Comment (kyrofa) [inline schema/snapcraft.yaml]: Should this be "no"?

File: snapcraft/tests/test_yaml.py [hunk 2]
```diff
@@ -734,6 +734,11 @@ def test_valid_app_daemons(self):
             'service3': {
                 'command': 'binary3',
                 'daemon': 'forking',
+            },
+            'service4': {
+                'command': 'binary4',
+                'daemon': 'simple',
+                'restart-condition': 'always',
```
Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: Worth adding tests for the other conditions?

**Comment by kyrofa** [comment]
Couple minor comments, but looks good :+1:

File: schema/snapcraft.yaml (hunk 1 above)
Comment (sergiusens) [inline schema/snapcraft.yaml]: The policy, as for `daemon` is to bring in the exact `systemd` names

File: snapcraft/tests/test_yaml.py (hunk 2 above)
Comment (sergiusens) [inline snapcraft/tests/test_yaml.py]: Sure

File: schema/snapcraft.yaml (hunk 1 above)
Comment (kyrofa) [inline schema/snapcraft.yaml]: Which at least according to [this](https://www.freedesktop.org/software/systemd/man/systemd.service.html) is "no," not "never." Right? Or am I reading something wrong?

File: snapcraft/tests/test_yaml.py (hunk 2 above)
Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: Not sure how important the watchdog test is... I guess it's possible to be added later. Maybe we don't want the brittleness there.

File: schema/snapcraft.yaml (hunk 1 above)
Comment (sergiusens) [inline schema/snapcraft.yaml]: I grabbed them from here https://github.com/ubuntu-core/snappy/blob/master/docs/meta.md
So more research is in order

File: schema/snapcraft.yaml [hunk 3]
```diff
@@ -168,6 +168,14 @@ properties:
...
               - simple
               - forking
+          restart-condition:
+              type: string
+              enum:
+                - on-success
+                - on-failure
+                - on-abnormal
+                - on-abort
+                - always
```
Comment (kyrofa) [inline schema/snapcraft.yaml]: Completely removing the "no?"

Comment (sergiusens) [inline schema/snapcraft.yaml]: Yes for now https://github.com/ubuntu-core/snappy/pull/567

**Pull request merged**